### PR TITLE
DBZ-2638 update OpenJDK quality outreach jobs

### PR DESCRIPTION
**Pull request by ani-sha**
https://issues.redhat.com/browse/DBZ-2638

### CODE REVIEW

**Comment by gunnarmorling** [comment]
@Naros, could you review this one? I'm a bit confused about the scope of _jobs.yml_ -- is this for _all_ jobs on GH Actions, or only for the Outreach jobs? When talking about the removal of the different MongoDB versions that are being tested, I was referring to the Outreach job only; of course we still need to keep version Matrix testing for the main job of the master branch. I'm not sure right now where this is happening, on GH, Travis, or even internal CI?

File: .github/workflows/jdk-outreach-workflow.yml [hunk 1]
```diff
@@ -117,53 +110,14 @@ jobs:
       with:
         java-version: ${{ matrix.java }}
     - name: Maven Build
-      run: mvn clean install -B -pl debezium-connector-mongodb -am -Passembly -Dformat.formatter.goal=validate
-        -Dformat.imports.goal=check -Dversion.mongo.server=3.2 -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn
-  mongodb-36:
```
Comment (gunnarmorling) [inline .github/workflows/jdk-outreach-workflow.yml:122]: So after discussing with @jpechane, I realize now this actually should remain as-is. I was under the impression that the MongoDB version matrix testing was related to the Outreach job, but turns out it isn't. So actually for MongoDB things are already as they ideally should be for all connectors/jobs: test all the required versions on GH Actions (provided its capacity and limits are enough for us to do so).

Comment (ani-sha) [reply .github/workflows/jdk-outreach-workflow.yml:122]: Alright. Will add them back.

**Comment by Naros** [comment]
@ani-sha, I'm curious if we could consolidate a few things here from each job into a `default` block as follows:
```
name: Java Quality Outreach
"on":
  schedule:
  - cron: 0 22 * * *
default:
  runs-on: ubuntu-latest
  strategy:
    matrix:
      java:
      - "15"
      - 16.0.0-ea.20
jobs:
 ...
```
The idea is that the `runs-on` and `strategy` sections for each job would get pulled into the `default` key/value pair map so that we don't need to manage these for each job itself.  What do you think?

**Comment by ani-sha** [comment]
> @ani-sha, I'm curious if we could consolidate a few things here from each job into a `default` block as follows:
> 
> ```
> name: Java Quality Outreach
> "on":
>   schedule:
>   - cron: 0 22 * * *
> default:
>   runs-on: ubuntu-latest
>   strategy:
>     matrix:
>       java:
>       - "15"
>       - 16.0.0-ea.20
> jobs:
>  ...
> ```
> 
> The idea is that the `runs-on` and `strategy` sections for each job would get pulled into the `default` key/value pair map so that we don't need to manage these for each job itself. What do you think? I'm not 100% sure this would work as I would like it to as the docs seem a tad vague, but you could test it locally in your fork by adding the job to fire on commits and just push some commits to a test branch for testing.

I think putting the `runs-on` and  `strategy` section into a `default` block makes sense, it would remove redundant parts.

**Comment by gunnarmorling** [comment]
Thanks for reviewing, @Naros. Leaving it to you to re-check and merge it once @ani-sha has addressed your comments. Thanks all!

**Comment by gunnarmorling** [comment]
@Naros, a re-review would be appreciated. Thanks!

**Review by Naros — APPROVED** [review]
LGTM, great work @ani-sha

**Pull request merged**